### ISSUE TITLE
Allow autocorrect to be configured by cop

### ISSUE THREAD

**Issue by jdoconnor** [bbatsov/rubocop#1626]
it would be great to be able to warn on a cop, but not autocorrect for it (maybe the cop is good, but not always valid is some situations)

something like the following in a rubocop.yml would be nice.

```yml
UnusedBlockArgument:
  AutoCorrect: false
```

**Comment by jdoconnor** [comment]
i'm happy to implement this if anyone thinks it would be usable

**Comment by bbatsov** [comment]
Generally our policy so far has been to improve problematic auto-corrections, but I can see the point in making it possible to disable them. @jonas054 Thoughts?

**Comment by jonas054** [comment]
There is a chance that people might turn off auto-correct in cops instead of reporting them, but I don't think that will happen a lot. So this feature could be useful, because there are other reasons than bugs why someone would want to turn off auto-correct per cop.

**Comment by bbatsov** [comment]
@jdoconnor Btw, what's wrong with `UnusedBlockArgument`'s autocorrect?

**Comment by jdoconnor** [comment]
just a preference in style for some situations. In the situation below, it would be helpful for grepping purposes to have the signatures (while read, not just executed) match

```
class OsxClass
  def foo(bar)
    raise NotImplementedError
  end
end

class Win32Class
  def foo(bar)
    bar = 42
  end
end
```

**Comment by bbatsov** [comment]
I'd suggest us fix #1601 then. We can simply add a check for `raise NotImplementedError` as well. :-)

Feel free to implement auto-correct disabling as well.
